example/kernelSpecialization/src/kernelSpecialization.cpp

```diff
@@ -23,18 +23,24 @@
 //#############################################################################
 //! Kernel to illustrate specialization for a particular accelerator
 //!
-//! It has a general operator() implementation and a specialized one for the CUDA accelerator.
-//! When running the kernel on a CUDA device, the specialized version of operator() is called.
-//! Otherwise the general version is called.
+//! It has a generic operator() implementation and an overload for the CUDA accelerator.
+//! When running the kernel on a CUDA device, the corresponding overload of operator() is called.
+//! Otherwise the generic version is called.
 //! The same technique can be applied for any function called from inside the kernel,
 //! thus allowing specialization of only relevant part of the code.
 //! It can be useful for optimization or accessing specific functionality not abstracted by alpaka.
+//!
+//! This kernel demonstrates the simplest way to achieve the effect by function overloading.
+//! Note that it does not perform specialization (in C++ template meaning) of function templates.
+//! We use word "specialization" as it represents a use case of having a special version for a particular accelerator.
```

Comment:
I think there is a *the* missing:

```suggestion
//! We use the word "specialization" as it represents a use case of having a special version for a particular accelerator.
```

Reply:
Fixed, added it to the last commit to not have too many commits.

Reply:
Perfect!